### PR TITLE
Dev/dmitrypa/daal soa vs homogen

### PR DESCRIPTION
**Pull request by dmitry741**
# Description
Speed up SOA format. The following case are covered:
1) AVX512 architecture
2) No data conversion

### CODE REVIEW

**Comment by dmitry741** [comment]
make test

File: include/data_management/data/internal/conversion.h [hunk 1]
```diff
@@ -66,9 +66,12 @@ inline ConversionDataType getConversionDataType<float>()
     return DAAL_SINGLE;
 }
 
-typedef void (*vectorConvertFuncType)(size_t n, const void * src, void * dst);
+typedef bool(*vectorCopy2vFuncType)(const size_t nrows, const size_t ncols, void* dst, void* ptrMin, int* arrOffests);
+typedef void(*vectorConvertFuncType)(size_t n, const void * src, void * dst);
+typedef void(*vectorStrideConvertFuncType)(size_t n, const void * src, size_t srcByteStride, void * dst, size_t dstByteStride);
 
-typedef void (*vectorStrideConvertFuncType)(size_t n, const void * src, size_t srcByteStride, void * dst, size_t dstByteStride);
+DAAL_EXPORT vectorCopy2vFuncType getVectorSingle();
```
Comment (ShvetsKS) [inline include/data_management/data/internal/conversion.h]: seems this and below func can be parameterized by template type parameter

**Comment by dmitry741** [comment]
make test

**Comment by dmitry741** [comment]
make test

**Comment by dmitry741** [comment]
make test

**Comment by dmitry741** [comment]
make test

**Comment by dmitry741** [comment]
make test

File: service/kernel/data_management/soa_numeric_table.cpp [hunk 2]
```diff
@@ -23,73 +23,82 @@ namespace data_management
...
+    if( memoryAllocationFlag == doAllocate )
     {
         st |= allocateDataMemoryImpl();
     }
 }
 
-} // namespace interface1
-} // namespace data_management
-} // namespace daal
+}
```
Comment (ShvetsKS) [inline service/kernel/data_management/soa_numeric_table.cpp]: revert deletion of namespaces.

**Comment by dmitry741** [comment]
make test

**Comment by ShvetsKS** [comment]
could you confirm that failed checks in precommit are not related to your changes and were already observed on master branch?

**Pull request closed without merging**